src/main/app/claims/models/claim.ts

```diff
@@ -411,6 +411,23 @@ export class Claim {
       && this.settlement && this.settlement.isOfferRejected()
   }
 
+  get isSettlementPaymentDateValid (): boolean {
+    if (this.settlement) {
+      const offer = this.settlement.getLastOffer()
+      const now = MomentFactory.currentDate()
+      if (offer && offer.paymentIntention && offer.paymentIntention.paymentOption === PaymentOption.BY_SPECIFIED_DATE) {
+        const paymentDate = offer.paymentIntention.paymentDate
+        return (paymentDate.isAfter(now) || paymentDate.isSame(now))
+      } else if (offer && offer.paymentIntention && offer.paymentIntention.paymentOption === PaymentOption.INSTALMENTS) {
```

Comment:
all of these if conditions are based off `offer && offer.paymentIntention` so we might extract that in a separate condition and then switch over offer.paymentIntention.paymentOption. I think it would be way more readable

Reply:
done